feature_engine/selection/shuffle_features.py

```diff
@@ -0,0 +1,213 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from sklearn.model_selection import train_test_split
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.metrics import roc_auc_score, mean_squared_error, r2_score
+
+from feature_engine.dataframe_checks import _is_dataframe, \
+    _check_input_matches_training_df
+
+from feature_engine.variable_manipulation import _define_variables, \
+    _find_all_variables
+
+
+class ShuffleFeatures(BaseEstimator, TransformerMixin):
+
+    """
+    
+    ShuffleFeatures reorganizes the values inside each feature, one feature 
+    at the time, from a dataframe and determines how that permutation affects
+    the performance metric of the machine learning algorithm.
+    
+    If the variables are important, a random permutation of their values will 
+    decrease dramatically any of these metrics. Contrarily, the permutation of 
+    values should have little to no effect on the model performance metric we 
+    are assessing.
+    
+
+    Parameters
+    ----------
+
+    features_to_shuffle : str or list, default=None
+        The list of variable(s) to be shuffled from the dataframe.
+        If None, the transformer will shuffle all variables in the dataset.
+        
+    estimator: object, default = DecisionTreeClassifier or DecisionTreeRegressor
+        depending on 'regression' value
+        Estimator can be a classifier or regressor
+        
+    scoring: str
+        metric used to evaluate to determine if feature needs to be kept or removed.
+
+    threshold: float
+        the value that defines if a feature will be kept or removed.
+
+    regression: boolean
+        Indicates whether the discretiser should train a regression or a classification
+        decision tree.
+
+    cv: boolean
+         Indicates whether cross-validation will be applied.
+    
+    """
+
+    def __init__(self,
+                 regression,
+                 cv = False,
+                 threshold = 0,
+                 features_to_shuffle = None,
+                 estimator = None,
+                 scoring = None):
+        
+        
+        if not isinstance(cv, bool):
+            raise ValueError("cv can only take True or False")
+
+        if not isinstance(regression, bool):
+            raise ValueError("regression can only take True or False")
+ 
+        self.features_to_shuffle = \
```

Comment:
we don't use \

If it doesn't fit in a line, black will auto-correct it to a style that will pass the tests